usr.sbin/adduser/adduser.sh

```diff
@@ -283,6 +288,11 @@ add_user() {
 		;;
 	esac
 
+	# create ZFS dataset if Zflag is set before home directory is created with pw
+	if [ -n "$Zflag" ]; then
```

Comment:
This should be negative (opt-out) instead of opt-in.

Probably:

```
if [ -n "$(zfs list -o name "${homeprefix}" 2>/dev/null)" ]; then
```

(the output _may_ be used as `zfs_homeprefix` too).